qiskit/dagcircuit/_dagnode.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+   Object to represent the information at a node in the DAGCircuit
+"""
+import copy
+
+
+class DAGNode:
+    """
+    Object to represent the information at a node in the DAGCircuit
+
+    It is used as the return value from *_nodes() functions and can
+    be supplied to functions that take a node.
+
+    """
+    def __init__(self, data_dict, nid=-1):
+        """ Create a node """
+        self._node_id = nid
+        self.data_dict = data_dict
+
+    @property
+    def type(self):
+        """ Returns a str which is the type of the node else None"""
+        return self.data_dict.get('type')
+
+    @property
+    def op(self):
+        """ Returns the Instruction object corresponding to the op for the node else None"""
+        return self.data_dict.get('op')
+
+    @property
+    def name(self):
+        """ Returns a str which is the name of the node else None"""
+        return self.data_dict.get('name')
+
+    @property
+    def qargs(self):
+        """
+        Returns list of (QuantumRegister, int) tuples where the int is the index
+        of the qubit else an empty list
+        """
+        return self.data_dict.get('qargs', [])
+
+    @property
+    def cargs(self):
+        """
+        Returns list of (ClassicalRegister, int) tuples where the int is the index
+        of the cbit else an empty list
+        """
+        return self.data_dict.get('cargs', [])
+
+    @property
+    def condition(self):
+        """ Returns a tuple (ClassicalRegister, int) where the int is the
+        value of the condition else None"""
+        return self.data_dict.get('condition')
+
+    @property
+    def wire(self):
+        """
+        Returns (Register, int) tuple where the int is the index of
+        the wire else None
+        """
+        return self.data_dict.get('wire')
+
+    def __getitem__(self, x):
+        return getattr(self, x)
```

Comment:
let's not have this method. it would encourage an "old style" lookup, which we are trying to move away from with the introduction of the DAGNode class.

Reply:
Ok will do, I just had it to make the change over easier.